Solar_System.rb

```diff
@@ -0,0 +1,166 @@
+require 'colorize'
+
+class SolarSystem
+
+  # Constractor
+  def initialize(planets)
+    @planets = planets
+  end
+
+  # Method who return the list of the planets with their properties and index
+  def list_planets
+    solar_system_details = ""
+    @planets.each_with_index do |planet, i|
+      solar_system_details += "#{i+1}. #{planet.get_description}\n"
+    end
+    return solar_system_details
+  end
+
+  # Method who return the planet according chosen index
+  def get_planet(index)
+    if index > 0 && index <= @planets.length
+      return @planets[index-1]
+    end
+  end
+
+  # Method who get a new planet and add it to the array
+  def add_planet(new_planet)
+    @planets << new_planet
+  end
+  
+end
+
+# my_planets = ["Mercury", "Venus", "Earth", "Mars", "Jupiter", "Hoth"]
+
+
+# my_planets = [
+# {name: "Mercury", diameter: "4,879", moons: "0", temperature: "-173-427", gravity: "3.7"},
+# {name: "Venus", diameter: "12,104", moons: "o", temperature: "462", gravity: "8.9},
+# {name: "Earth", diameter: "12,756", moons: "1", temperature: "-88 to 58", gravity: "9.8"},
+# {name: "Mars", diameter: "6,805", moons: "2", temperature: "-63", gravity: "3.7"},
+# {name: "Jupiter", diameter: "142,984", moons: "67", temperature: "-108", gravity: "24.8"},
+# {name: "Hoth", diameter: "7,200", moons: "3", temperature: "-60", gravity: "1.1"}
+# ]
+
+
+class Planet
+
+  # Constractor
+  def initialize(name, diameter, moons, temperature, gravity)
+    @name = name
+    @diameter = diameter
+    @moons = moons
+    @temperature = temperature
+    @gravity = gravity
+  end
+
+  # Method who return the planet description
+  def get_description
+    return "name: #{@name}, Diameter: #{@diameter}, Moons: #{@moons}, Temperature: #{@temperature}, Gravity: #{@gravity}".green
+  end
+
+  # Getter for name properties
+  def read_name
+    return @name
```

Comment:
Instead of defining `read_name` here, you should use an `attr_reader`:

```ruby
class Planet
  attr_reader :name, :diameter, :moons, :temperature, :gravity
  def initialize(name, diameter, moons, temperature, gravity)
    @name = name
    # ...
  end
end
```

From outside, instead of `planet.read_name` you would say `planet.name`.